src/Stratis.Bitcoin/P2P/RelatedPeerConnectors.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+
+namespace Stratis.Bitcoin.P2P
+{
+    public sealed class RelatedPeerConnectors : Dictionary<string, PeerConnector>
```

Comment:
As I said I recommend we remove this class all together and use the ConnectionManager.ConnectedPeers inside each connector.